channels/layers.py

```diff
@@ -155,19 +159,13 @@ def valid_channel_name(self, name, receive=False):
                         "Specific channel names in receive() must end at the !"
                     )
                 return True
-        raise TypeError(
-            "Channel name must be a valid unicode string containing only ASCII "
-            + "alphanumerics, hyphens, or periods, not '{}'.".format(name)
-        )
+        raise TypeError(self.invalid_name_error.format("Channel", name))
 
     def valid_group_name(self, name):
         if self.match_type_and_length(name):
             if bool(self.group_name_regex.match(name)):
                 return True
-        raise TypeError(
-            "Group name must be a valid unicode string containing only ASCII "
-            + "alphanumerics, hyphens, or periods."
-        )
+        raise TypeError(self.invalid_name_error.format("Group", name))
```

Comment:
I wonder why it's a TypeError and not a ValueError, @andrewgodwin do you have a clue about that perhaps ?